### PR TITLE
Updated docs for postgres and mysql components

### LINKED ISSUE #3602: [1.12] Update Postgres components reference
Update the Postgres components reference: state store, binding, config store.

Need to add:

- Support for Azure AD
- New/updated metadata properties
- Support for parametrized queries in binding

Code PRs:

- dapr/components-contrib#2970
- dapr/components-contrib#2971
- dapr/components-contrib#2972

### PR DESCRIPTION
**Pull request by ItalyPaleAle**
Fixes #3602

Also includes docs for https://github.com/dapr/components-contrib/pull/2975

### CODE REVIEW

File: daprdocs/content/en/reference/components-reference/supported-bindings/mysql.md [hunk 1]
```diff
@@ -81,17 +87,37 @@ This component supports **output binding** with the following operations:
...
+```sql
+-- ❌ WRONG! Includes values in the query and is vulnerable to SQL Injection attacks.
+SELECT * FROM mytable WHERE user_key = 'something';
+
+-- ✅ GOOD! Uses parametrized queries.
+-- This will be executed with parameters ["something"]
+SELECT * FROM mytable WHERE user_key = ?;
+```
+
+> [Relevant XKCD](https://xkcd.com/327/)
```
Comment (ItalyPaleAle) [inline daprdocs/content/en/reference/components-reference/supported-bindings/mysql.md]: My .02. this is well-known enough, and humor helps getting important points across!

File: daprdocs/content/en/reference/components-reference/supported-bindings/mysql.md [hunk 2]
```diff
@@ -55,21 +55,27 @@ Note that you can not use secret just for username/password. If you use secret,
 | `maxOpenConns` | N | Output | The max open connections. Integer greater than 0 | `"10"` |
 | `connMaxLifetime` | N | Output | The max connection lifetime. Duration string | `"12s"` |
 | `connMaxIdleTime` | N | Output | The max connection idel time. Duration string | `"12s"` |
-| `direction` | N | Output | The direction of the binding | `"output"` |
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/mysql.md]: Can we keep the `direction` on this component. We put these into all binding to be explicit that this is useful for separation of Dapr sidecar from asking the app.

File: daprdocs/content/en/reference/components-reference/supported-bindings/mysql.md (hunk 1 above)
Comment (artursouza) [reply daprdocs/content/en/reference/components-reference/supported-bindings/mysql.md]: That is funny and will be recorded in the history of this PR. Can we remove it to get this merged?

File: daprdocs/content/en/reference/components-reference/supported-bindings/mysql.md [hunk 3]
```diff
@@ -55,21 +55,27 @@ Note that you can not use secret just for username/password. If you use secret,
...
 
 If your server requires SSL your connection string must end of `&tls=custom` for example:
+
 ```bash
 "<user>:<password>@tcp(<server>:3306)/<database>?allowNativePasswords=true&tls=custom"
 ```
- You must replace the `<PEM PATH>` with a full path to the PEM file. If you are using [MySQL on Azure](http://bit.ly/AzureMySQLSSL) see the Azure [documentation on SSL database connections](http://bit.ly/MySQLSSL), for information on how to download the required certificate. The connection to MySQL will require a minimum TLS version of 1.2.
 
-Also note that by default [MySQL go driver](https://github.com/go-sql-driver/mysql) only supports one SQL statement per query/command.
+> You must replace the `<PEM PATH>` with a full path to the PEM file. If you are using Azure Database for MySQL see the Azure [documentation on SSL database connections](https://learn.microsoft.com/azure/mysql/single-server/how-to-configure-ssl), for information on how to download the required certificate. The connection to MySQL will require a minimum TLS version of 1.2.
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/mysql.md]:
```suggestion
> You must replace the `<PEM PATH>` with a full path to the PEM file. If you are using Azure Database for MySQL see the Azure [documentation on SSL database connections](https://learn.microsoft.com/azure/mysql/single-server/how-to-configure-ssl), for information on how to download the required certificate. The connection to MySQL requires a minimum TLS version of 1.2.
```

**Review by artursouza — APPROVED** [review]

**Review by hhunter-ms — APPROVED** [review]

File: daprdocs/content/en/reference/components-reference/supported-bindings/postgresql.md [hunk 4]
```diff
@@ -22,10 +22,9 @@ spec:
   type: bindings.postgresql
   version: v1
   metadata:
-  - name: url # Required
-    value: "<CONNECTION_STRING>"
-  - name: direction
-    value: "<DIRECTION_OF_BINDING>"
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/postgresql.md:28]: Please include the DIRECTION_OF_BINDING

Comment (ItalyPaleAle) [reply daprdocs/content/en/reference/components-reference/supported-bindings/postgresql.md:28]: This is only an output binding so the direction is irrelevant. I think it causes more confusion than anything.

File: daprdocs/content/en/reference/components-reference/supported-bindings/mysql.md (hunk 2 above)
Comment (ItalyPaleAle) [reply daprdocs/content/en/reference/components-reference/supported-bindings/mysql.md]: This is only an output binding so the direction is irrelevant. I think it causes more confusion than anything.

Comment (msfussell) [reply daprdocs/content/en/reference/components-reference/supported-bindings/mysql.md]: It still tells the sidecar information that is needed. Even for output only bindings this is useful, yes?

Comment (ItalyPaleAle) [reply daprdocs/content/en/reference/components-reference/supported-bindings/mysql.md]: No, it is a no-op for output bindings, just adds confusion to users

Comment (ItalyPaleAle) [reply daprdocs/content/en/reference/components-reference/supported-bindings/mysql.md]: I am referring specifically to output-only bindings. A binding that is input-only or input and output can benefit from this. But output bindings don't need this

**Pull request merged**